media_kit/lib/src/models/media.dart

```diff
@@ -46,58 +47,59 @@ class Media extends Playable {
   static String normalizeURI(String uri) {
     if (uri.startsWith(_kAssetScheme)) {
       // Handle asset:// scheme. Only for Flutter.
+      final key = encodeAssetKey(uri);
+      final String asset;
       if (Platform.isWindows) {
-        return Uri.file(
+        asset = path.normalize(
           path.join(
             path.dirname(Platform.resolvedExecutable),
             'data',
             'flutter_assets',
-            path.normalize(uri.split(_kAssetScheme).last),
+            key,
           ),
-        ).toString();
-      }
-      if (Platform.isLinux) {
-        return Uri.file(
+        );
+      } else if (Platform.isLinux) {
+        asset = path.normalize(
           path.join(
             path.dirname(Platform.resolvedExecutable),
             'data',
             'flutter_assets',
-            path.normalize(uri.split(_kAssetScheme).last),
+            key,
           ),
-        ).toString();
-      }
-      if (Platform.isMacOS) {
-        return Uri.file(
+        );
+      } else if (Platform.isMacOS) {
+        asset = path.normalize(
           path.join(
             path.dirname(Platform.resolvedExecutable),
             '..',
             'Frameworks',
             'App.framework',
             'Resources',
             'flutter_assets',
-            path.normalize(uri.split(_kAssetScheme).last),
+            key,
           ),
-        ).toString();
-      }
-      if (Platform.isIOS) {
-        return Uri.file(
+        );
+      } else if (Platform.isIOS) {
+        asset = path.normalize(
           path.join(
             path.dirname(Platform.resolvedExecutable),
             'Frameworks',
             'App.framework',
             'flutter_assets',
-            path.normalize(uri.split(_kAssetScheme).last),
+            key,
           ),
-        ).toString();
-      }
-      if (Platform.isAndroid) {
-        return AndroidAssetLoader.instance.load(
-          path.normalize(uri.split(_kAssetScheme).last),
         );
+      } else if (Platform.isAndroid) {
+        asset = path.normalize(AndroidAssetLoader.instance.load(key));
+      } else {
+        throw UnimplementedError(
+          '$_kAssetScheme is not supported on ${Platform.operatingSystem}',
+        );
+      }
+      if (!File(asset).existsSync_()) {
+        throw Exception('Unable to load asset: $asset');
```

Comment:
Now throwing an exception if an asset is not located.

![](https://user-images.githubusercontent.com/28951144/231520936-d21b06f1-4415-4096-9315-bc5113bdc224.png)